packages/manager/apps/telecom/src/app/telecom/telephony/alias/portability/portabilities/portabilities.service.js

```diff
@@ -1,18 +1,91 @@
-import find from 'lodash/find';
-import forEach from 'lodash/forEach';
+export default class TelecomTelephonyAliasPortabilitiesService {
+  /* @ngInject */
+  constructor($http, $q, iceberg) {
+    this.$http = $http;
+    this.$q = $q;
+    this.iceberg = iceberg;
+  }
 
-export default function groupPortaByNumbers(portabilities) {
-  const numbers = [];
-  forEach(portabilities, (portability) => {
-    forEach(portability.numbersList, (number) => {
-      numbers.push({
-        number,
-        portability,
-        lastStepDone: find(portability.steps.slice().reverse(), {
-          status: 'done',
-        }),
+  fetchPortability(billingAccount) {
+    return this.iceberg(`/telephony/${billingAccount}/portability`)
+      .query()
+      .expand('CachedObjectList-Pages')
+      .sort('creationDate', 'ASC')
+      .execute()
+      .$promise.then(({ data: portabilities }) => {
+        return this.$q.all(
+          portabilities.map((porta) => {
+            return this.$q
+              .all({
+                steps: this.portabilityStatus(billingAccount, porta.id),
+                canBeCancelled: this.canBeCancelled(billingAccount, porta.id),
+                documentAttached: this.documentAttached(
+                  billingAccount,
+                  porta.id,
```

Comment:
This looks like a lot of calls if cutomer has a lot of portabilities ongoing, are we sure we need everything at once ?

Reply:
It was already like that today.
The portabilities are managed group by group in the manager. This limits the number.